index.js

```diff
@@ -115,28 +133,44 @@ module.exports = function (){
 
     this.prepareSignSend = function(abi,contractAddress,functionName,senderAddress,privateKey, params, gasLimit){
         return new Promise(async (resolve, reject) => {
-
-            var txnData = this.encodeFunctionParams(abi, functionName,  params);
-            var _gasLimit = await this.ContractInstance.methods[functionName](...params).estimateGas({'from': senderAddress, 'gas': gasLimit});
-            var txnRawData = await this.getDefaultTxnAttributes('',senderAddress,contractAddress,'0',txnData,_gasLimit,'')
-            var dataToSend = this.getSignedTransaction(txnRawData, privateKey);
-
-            await this.invokeSendRawTransaction(functionName,dataToSend).then((result) =>{
-                resolve(result);
-            },(error) =>{
-                reject(error);
-            });
+            try{
+                var txnData = this.encodeFunctionParams(abi, functionName,  params);
+                var _gasLimit = await this.ContractInstance.methods[functionName](...params).estimateGas({'from': senderAddress, 'gas': gasLimit});
+                var txnRawData = await this.getDefaultTxnAttributes('',senderAddress,contractAddress,'0',txnData,_gasLimit,'')
+                if (txnRawData) {
+                    var dataToSend = this.getSignedTransaction(txnRawData, privateKey);
+                    web3.eth.sendSignedTransaction(dataToSend, function(error, txHash) {
```

Comment:
This can be try catched, and since you only push out the message which is the error, you dont need to do the callback way to get granularity on this part

Reply:
Fixed, but unfortunately I cannot get just the transaction hash now. It waits till the receipt is ready. Tried the event emitter approach (with .on) but it doesn't work it seems.